src/idom/utils.py

```diff
@@ -233,9 +232,9 @@ def _mutate_vdom(vdom: VdomDict) -> None:
         # Convince type checker that it's safe to mutate attributes
         assert isinstance(vdom["attributes"], dict)
 
-        # Convert style attribute from str -> dict with camelCase keys
+        # Convert style attribute from str -> dict with snake case keys
         vdom["attributes"]["style"] = {
-            _hypen_to_camel_case(key.strip()): value.strip()
+            key.strip().replace("-", "_"): value.strip()
```

Comment:
I think we should assume that all dashes (`-`) are intentional.

Reply:
We actually need to do this. Previously we converted directly to camelCase because this is what React expects client-side. Now though, purely for the sake of consistency we convert to snake_case, which the client subsequently converts to camelCase.

Reply:
Perhaps I'm not understanding, but I don't see any reason why we couldn't directly communicate props with dashes (`-`) to React.

Seems to be [a valid prop value](https://stackoverflow.com/questions/27285856/how-do-i-dynamically-set-html5-data-attributes-using-react)?

Reply:
This conversion happens to only impact the `style` dict however, in general, we [handle](https://github.com/idom-team/idom/blob/ddc7178c7bbf8efcedac5c6ac43308eb059504bd/src/client/packages/idom-client-react/src/element-utils.js#L54-L78) data/aria attributes in the client. Specifically, the behavior is, if your attribute name begins with `data_` or `aria_`, we convert all `_` to `-` characters.